packages/financial-templates-lib/src/helpers/GasEstimator.ts

```diff
@@ -251,19 +251,19 @@ export class GasEstimator {
 
   private _extractFastGasPrice(json: { [key: string]: any }, url: string): LondonGasData | LegacyGasData {
     if (url.includes("etherscan.io") || url.includes("polygonscan.com")) {
+      const isMainnet = url.includes("etherscan.io");
       const etherscanGasResponse = json as EtherscanGasResponse;
       if (etherscanGasResponse.result.suggestBaseFee === undefined)
         throw new Error(`Bad ethgasstation response ${json}`);
       return {
-        maxFeePerGas: Number(etherscanGasResponse.result.FastGasPrice),
-        maxPriorityFeePerGas:
-          Number(etherscanGasResponse.result.FastGasPrice) - Number(etherscanGasResponse.result.suggestBaseFee),
+        maxFeePerGas: Number(etherscanGasResponse.result.suggestBaseFee) * 3,
+        maxPriorityFeePerGas: isMainnet ? 5 : 50,
```

Comment:
5 is really high for priority fee. A low priority fee will _only not be included_ if the block is 100% full of txns with a higher priority fee (which causes the base fee to increase). I would set this to 1.

For polygon, they apparently don't implement EIP 1559 "correctly", so it does need to be set fairly high. 50 is probably fine, but I've read that you really do need to use a gas estimating service to really get a good number. Since we probably don't send a lot of polygon transfers from these bots, overpricing on polygon is probably fine.

Reply:
Thanks @mrice32, set mainnet maxPriorityFeePerGas to 1 in latest commit